src/harness/reference_models/interference/interference.py

```diff
@@ -413,45 +413,67 @@ def computeInterferenceFssCochannel(cbsd_grant, constraint, fss_info, max_eirp):
 
-  fss_mask_loss = 0
-
-  # if lower edge of the FSS passband is less than CBSD grant
-  # lowFrequency and highFrequency
-  if (constraint.low_frequency < cbsd_grant.low_frequency and
-         constraint.low_frequency < cbsd_grant.high_frequency):
-    fss_mask_loss = 0.5
-
-  # if CBSD grant lowFrequency and highFrequency is less than
-  # 50MHz offset from the FSS passband lower edge
-  elif (cbsd_grant.low_frequency < offset and
-     cbsd_grant.high_frequency < offset):
-    fss_mask_loss = linearToDb((cbsd_freq_range / MHZ) * 0.25)
-
-  # if CBSD grant lowFrequency is less than 50MHz offset and
-  # highFrequency is greater than 50MHz offset
-  elif (cbsd_grant.low_frequency < offset and
-            cbsd_grant.high_frequency > offset):
-    low_freq_mask_loss = linearToDb(((offset - cbsd_grant.low_frequency) / MHZ) * 0.25)
-    fss_mask_loss = low_freq_mask_loss + linearToDb(((
-        cbsd_grant.high_frequency - offset) / MHZ) * 0.6)
-
-  # if FSS Passband lower edge frequency is grater than CBSD grant
-  # lowFrequency and highFrequency and
-  # CBSD grand low and high frequencies are greater than 50MHz offset
-  elif (constraint.low_frequency > cbsd_grant.low_frequency and
-      constraint.low_frequency > cbsd_grant.high_frequency and
-      cbsd_grant.low_frequency > offset and
-             cbsd_grant.high_frequency > offset):
-    fss_mask_loss = linearToDb((cbsd_freq_range / MHZ) * 0.6)
-
-  return fss_mask_loss
+  # Define the initial attenuation values to 0
+  fss_mask_attenuation = 0
+  fss_mask_attenuation_seg1 = 0
+  fss_mask_attenuation_seg2 = 0
+  
+  # If the grant overlap frequency lies between 3650 to 3700 MHz
+  # Segment 1 calculation is applied
+  if ( offset <= cbsd_grant.low_frequency and
+         constraint.high_frequency >= cbsd_grant.high_frequency):
+   # Calculate the total of 1MHZ frequencies present in the grant frequency range
+   num_one_mhz_bin_seg1 = cbsd_freq_range/int(MHZ)
+   # Calculate the fss_mask for every 1MHZ frequency 
+   for i in range (0, num_one_mhz_bin_seg1 - 1):
+     fss_mask_attenuation += 10 **((dbToLinear(-(constraint.high_frequency - cbsd_grant.low_frequency)  - 0.5 -i)*0.6 - 0.5)/10) 
+
+   # Calculate the average attenuation value per MHZ
+   fss_mask_attenuation_seg1 = linearToDb(fss_mask_attenuation/num_one_mhz_bin_seg1)
+
+  # If the grant overlap frequency lies between 3550 to 3650 MHz
+  # Segment 2 calculation is applied
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+          offset >= cbsd_grant.high_frequency):
+    fss_mask_attenuation = 0
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg2 = cbsd_freq_range/int(MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg2 - 1):
+      fss_mask_attenuation += 10 **((dbToLinear(-(offset - cbsd_grant.low_frequency)  - 0.5 -i)*0.25 - 30.5)/10)
+    # Calculate the average attenuation value per MHZ
```

Comment:
Same comments as above for the segment calculation: use range (0,num_one_mhz_bin_seg2 ) and the braces do not look right